pyright_reportGeneralTypeIssues.json

```diff
@@ -57,58 +56,46 @@
         "pandas/core/indexes/extension.py",
         "pandas/core/indexes/interval.py",
         "pandas/core/indexes/multi.py",
-        "pandas/core/indexes/numeric.py",
         "pandas/core/indexes/period.py",
-        "pandas/core/indexes/timedeltas.py",
         "pandas/core/indexing.py",
         "pandas/core/internals/api.py",
         "pandas/core/internals/array_manager.py",
         "pandas/core/internals/blocks.py",
-        "pandas/core/internals/concat.py",
         "pandas/core/internals/construction.py",
         "pandas/core/internals/managers.py",
         "pandas/core/missing.py",
         "pandas/core/nanops.py",
+        "pandas/core/ops/array_ops.py",
```

Comment:
This gives too many issues to fix here, is that right? Same with json/normalize below.

Reply:
There are many files with only 1-2 violations (including the two added in this PR): we could simply add inline ignores for them and remove these files from this list.

Reply:
While I think it would be beneficial to add pyright-specific inline ignores (so that mypy doesn't call them out as unneeded) we currently have no way for pyright to flag unneeded ignore comments without pyright also calling out the general-ignore comments that mypy needs but pyright may not.

I'm inclined to prefer the file-based ignore list until we have a check to flag unneeded pyright-specific ignores. I think there are three options:

1. Let pyright ignore the general comments, add many(!) pyright specific ignores to the code base (some lines will have mypy and pyright ignores!), enable pyright's check for unused comments. Pandas-stubs does that.
2. Have a third pyright run that checks for reportGeneralTypeIssues and unneeded comments for it but we filter out the reportGeneralTypeIssues violations
3. Ask pyright to have an option to check only whether its specific ignore comments are unused (there is already an issue at mypy to have mypy-specific comments - unlikely to happen).